libcore/src/geometry/Building.cpp

```diff
@@ -286,7 +289,13 @@ void Building::AddSurroundingRoom()
 bool Building::InitGeometry()
 {
     Logging::Info("Init Geometry");
-    correct();
+    try {
+        geometry::helper::correct(*this);
```

Comment:
Can we rename this to be more expressive?
E.g.:
```
correctInputGeometry()
```

Reply:
done